core/network/src/consensus_gossip.rs

```diff
@@ -112,36 +119,22 @@ impl<B: BlockT> ConsensusGossip<B> {
 	}
 
 	/// Handle new connected peer.
-	pub fn new_peer(&mut self, protocol: &mut Context<B>, who: NodeIndex, roles: Roles) {
-		if roles.intersects(Roles::AUTHORITY) {
-			trace!(target:"gossip", "Registering {:?} {}", roles, who);
-			let now = Instant::now();
-			// Send out all known messages to authorities.
-			let mut known_messages = HashSet::new();
-			for entry in self.messages.iter() {
-				if entry.timestamp + MESSAGE_LIFETIME < now { continue }
-				if let Status::Future = entry.status { continue }
-
-				known_messages.insert(entry.message_hash);
-				protocol.send_message(who, Message::Consensus(entry.message.clone()));
-			}
-			self.peers.insert(who, PeerConsensus {
-				known_messages,
-				is_authority: true,
-			});
-		}
-		else if roles.intersects(Roles::FULL) {
-			self.peers.insert(who, PeerConsensus {
-				known_messages: HashSet::new(),
-				is_authority: false,
-			});
+	pub fn new_peer(&mut self, _protocol: &mut Context<B>, who: NodeIndex, roles: Roles) {
+		trace!(target:"gossip", "Registering {:?} {}", roles, who);
+		self.peers.insert(who, PeerConsensus {
+			known_messages: HashSet::new(),
+			is_authority: roles.intersects(Roles::AUTHORITY),
```

Comment:
I would like to move this down to the gossip layer at some point. Basically the gossip validator is asked `propagate_message_to(peer, message) -> Propagate` where `Propagate` has these variants:
  - `Yes` (this is the behavior we currently have for authorities).
  - `IfSampled` (this is the behavior we currently have for non-authorities).
  - `No` (this is a behavior we are introducing shortly)

Reply:
There is no sampling currently, and the gossip now makes no distinction between authorities and non-authorities, other than logging messages. We might add sampling in the future though.

Reply:
We could get rid of the `is_authority` thing then

Reply:
My mistake. The flag is still used from propagation. I'm going to remove it.